### ISSUE TITLE
Password's can't have "<" in them

### ISSUE THREAD

**Issue by JonnyAxehandle** [voat/voat#544]
Passwords, assuming they are being properly encrypted, should not impose arbitrary limits on characters like this

**Comment by Aaron-P** [comment]
Is there a specific error? That sort of thing might be hitting an html filter and need [AllowHtml] or something.

**Comment by TNTGav** [comment]
This is blocked explicitly but I'm not sure why:

AccountViewModels.cs, line 37 =>

```c#
[RegularExpression("^[^<]+$", ErrorMessage = "The character < is not allowed. Sorry.")]
```

It may have been written in to stop something breaking, as it has been explicitly written in I'd caution removing it without knowing why it was added in the first place.

**Comment by talons1337** [comment]
It's very obviously there to prevent code injections. < is the opening tag for a HTML element.

**Comment by Aaron-P** [comment]
Except that properly implemented password systems don't expose the password field in any way, and thus cannot be injected.

**Comment by faissaloo** [comment]
I agree, preventing the '<' character from being used in a password is a bit too 'hacky' tbh.

**Comment by auscompgeek** [comment]
For reference, this was introduced in 41d36a024e1b25224c5d434ea229b2e000e71852.

**Comment by linusbergl** [comment]
This is also a bother when using password managers to generate passwords. I generated a 50 character random password and got the error it wasn't secure. Ffor reference, it was something like 290 bits security. More than sufficiently strong for launching nuclear missiles, but apparently not good enough for voat. I got the error that I didn't use any special characters, and when I enabled them I got the error that I was using <.

This seems rather arbitrary. The password should of course be hashed, so that restriction makes no sense.
